SSENG/View/MapView/MapViewController.swift

```diff
@@ -113,18 +248,74 @@ class MapViewController: UIViewController {
     navigationController?.setNavigationBarHidden(false, animated: false)
   }
 
-  // 뷰 추가
+  // MARK: - 뷰 추가
+
   private func setupUI() {
-    [mapView, controlStackView, myPageButton, rideKickBoardView].forEach { view.addSubview($0) }
-    [reloadButton, dividerView, locationButton].forEach { controlStackView.addArrangedSubview($0) }
+    [mapView, myPageButton, markerFilterStackView, controlStackView, rideKickBoardView].forEach { view.addSubview($0) }
+    [allMarkerButton, dividerView1, kickBoardMarkerButton, dividerView2, bikeMarkerButton, dividerView3, noneMarkerButton].forEach { markerFilterStackView.addArrangedSubview($0) }
+    [reloadButton, dividerView4, locationButton].forEach { controlStackView.addArrangedSubview($0) }
+    [kickBoardHStackView, rideingButton].forEach { rideKickBoardView.addSubview($0) }
+    [typeImageView, kickBoardVStackView].forEach { kickBoardHStackView.addArrangedSubview($0) }
+    [batteryLabel, priceLabel, detailLocationTitleLabel, detailLocationLabel].forEach { kickBoardVStackView.addArrangedSubview($0) }
+
+    allMarkerButton.addSubview(allMarkerIndicatorDot)
+    kickBoardMarkerButton.addSubview(kickBoardIndicatorDot)
+    bikeMarkerButton.addSubview(bikeIndicatorDot)
+    noneMarkerButton.addSubview(noneIndicatorDot)
   }
 
-  // 제약조건
+  // MARK: - 제약조건
+
   private func setupConstraints() {
     mapView.snp.makeConstraints {
       $0.directionalEdges.equalToSuperview()
     }
 
+    myPageButton.snp.makeConstraints {
+      $0.trailing.equalToSuperview().inset(20)
+      $0.top.equalTo(view.safeAreaLayoutGuide).offset(20)
+      $0.size.equalTo(50)
+    }
+
+    markerFilterStackView.snp.makeConstraints {
+      $0.trailing.equalToSuperview().inset(20)
+      $0.bottom.equalTo(controlStackView.snp.top).offset(-10)
+      $0.width.equalTo(50)
+    }
+
+    allMarkerButton.snp.makeConstraints {
+      $0.height.equalTo(50)
+    }
+
+    kickBoardMarkerButton.snp.makeConstraints {
+      $0.height.equalTo(50)
+    }
+
+    bikeMarkerButton.snp.makeConstraints {
+      $0.height.equalTo(50)
+    }
+
+    noneMarkerButton.snp.makeConstraints {
+      $0.height.equalTo(50)
+    }
```

Comment:
제약 조건이 동일하고, 이후에 버튼이 추가될 가능성이 없다면 forEach문을 사용해서 중복을 줄이는 방식도 괜찮지 않을까 해서 권유드려봅니다 :)

```suggestion
[allMarkerButton, kickBoardMarkerButton, bikeMarkerButton, noneMarkerButton].forEach {
  $0.snp.makeConstraints {
    $0.height.equalTo(50)
  }
}
```

아니면 이런 방식도 좋지 않을까 싶은 생각입니다
```swift
let heightFixedViews = [
  allMar...,
  kick...,
  bike...,
  noneMarkerButton,
  ...
]

heightFixedViews.forEach {
  $0.snp.makeConstraints {
    $0.height.equalTo(50)
  }
}
```

Reply:
감사합니다! 추후 리팩토링을 진행해 해 보겠습니다!